Woops, something's gone wrong! The CardDAV server returned the http status code 500.

I use sabre-dav in combination with z-push (http://z-push.org/).
Since the update of sabre-dav from 3.2.0 to 3.2.1, z-push is broken and I get error messages like the following in the z-push log:

```
06/12/2017 22:47:48 [ 5484] [ INFO] [unknown] ExportChangesDiff->InitializeExporter(): Found '1' changes for 'contacts'
06/12/2017 22:47:49 [ 5484] [ERROR] [unknown] BackendCardDAV->GetMessage - Error getting vcard 'a4de694e-60c3-4a50-b504-bd2641d3b2a4' in '7f7297f2': Woops, something's gone wrong! The CardDAV server returned the http status code 500.
06/12/2017 22:47:49 [ 5484] [ERROR] [unknown] BackendCardDAV->GetMessage(): vCard not found
06/12/2017 22:47:49 [ 5484] [WARN] [unknown] C:\src\z-push\src\lib\default\diffbackend\exportchangesdiff.php:163 Creating default object from empty value (2)
```
Running a git bisec points at 73065c3273cc517e201354b06d3918c88d755d99 as the first faulty commit. This was added for #889

--- a/lib/CardDAV/Xml/Filter/AddressData.php
+++ b/lib/CardDAV/Xml/Filter/AddressData.php
@@ -52,12 +52,15 @@ class AddressData implements XmlDeserializable {
         ];
 
         $elems = (array)$reader->parseInnerTree();
+        $elems = array_filter($elems, function($element) {
+            return $element['name'] === '{urn:ietf:params:xml:ns:carddav}prop' &&
+                isset($element['attributes']['name']);
+        });
         $result['addressDataProperties'] = array_map(function($element) {
             return $element['attributes']['name'];
         }, $elems);
 
         return $result;
-
     }
 
 }

--- a/tests/Sabre/CardDAV/Xml/Request/AddressBookMultiGetTest.php
+++ b/tests/Sabre/CardDAV/Xml/Request/AddressBookMultiGetTest.php
@@ -10,21 +10,13 @@ class AddressBookMultiGetTest extends XmlTest {
         '{urn:ietf:params:xml:ns:carddav}addressbook-multiget' => 'Sabre\\CardDAV\\Xml\\Request\AddressBookMultiGetReport',
     ];
 
-    function testDeserialize() {
+    /**
+     * @dataProvider providesAddressDataXml
+     * @param $xml
+     */
+    function testDeserialize($xml, $expectedProps, $expectedVersion = '3.0') {
 
         /* lines look a bit odd but this triggers an XML parsing bug */
-        $xml = <<<XML
-<?xml version='1.0' encoding='UTF-8' ?>
-<CARD:addressbook-multiget xmlns:d="DAV:" xmlns:CARD="urn:ietf:params:xml:ns:carddav">
-  <d:prop>
-    <d:getcontenttype />
-    <d:getetag />
-    <CARD:address-data content-type="text/vcard" version="4.0" />
-  </d:prop>
-  <d:href>/foo.vcf</d:href>
-</CARD:addressbook-multiget>
-XML;
-
         $result = $this->parse($xml);
         $addressBookMultiGetReport = new AddressBookMultiGetReport();
         $addressBookMultiGetReport->properties = [
@@ -34,8 +26,8 @@ XML;
         ];
         $addressBookMultiGetReport->hrefs = ['/foo.vcf'];
         $addressBookMultiGetReport->contentType = 'text/vcard';
-        $addressBookMultiGetReport->version = '4.0';
-        $addressBookMultiGetReport->addressDataProperties = [];
+        $addressBookMultiGetReport->version = $expectedVersion;
+        $addressBookMultiGetReport->addressDataProperties = $expectedProps;
 
 
         $this->assertEquals(
@@ -45,5 +37,53 @@ XML;
 
     }
 
-
+    function providesAddressDataXml()
+    {
+        $simpleXml = <<<XML
+<?xml version='1.0' encoding='UTF-8' ?>
+<CARD:addressbook-multiget xmlns:d="DAV:" xmlns:CARD="urn:ietf:params:xml:ns:carddav">
+  <d:prop>
+    <d:getcontenttype />
+    <d:getetag />
+    <CARD:address-data content-type="text/vcard" version="4.0"/>
+  </d:prop>
+  <d:href>/foo.vcf</d:href>
+</CARD:addressbook-multiget>
+XML;
+        $allPropsXml = <<<XML
+<?xml version='1.0' encoding='UTF-8' ?>
+<CARD:addressbook-multiget xmlns:d="DAV:" xmlns:CARD="urn:ietf:params:xml:ns:carddav">
+  <d:prop>
+    <d:getcontenttype />
+    <d:getetag />
+    <CARD:address-data>
+        <CARD:allprop/>
+    </CARD:address-data>
+  </d:prop>
+  <d:href>/foo.vcf</d:href>
+</CARD:addressbook-multiget>
+XML;
+        $multiplePropsXml = <<<XML
+<?xml version='1.0' encoding='UTF-8' ?>
+<CARD:addressbook-multiget xmlns:d="DAV:" xmlns:CARD="urn:ietf:params:xml:ns:carddav">
+  <d:prop>
+    <d:getcontenttype />
+    <d:getetag />
+    <CARD:address-data>
+        <CARD:prop name="VERSION"/>
+        <CARD:prop name="UID"/>
+        <CARD:prop name="NICKNAME"/>
+        <CARD:prop name="EMAIL"/>
+        <CARD:prop name="FN"/>
+    </CARD:address-data>
+  </d:prop>
+  <d:href>/foo.vcf</d:href>
+</CARD:addressbook-multiget>
+XML;
+        return [
+            'address data with version'         => [$simpleXml, [], '4.0'],
+            'address data with inner all props' => [$allPropsXml, []],
+            'address data with mutliple props'  => [$multiplePropsXml, ['VERSION', 'UID', 'NICKNAME', 'EMAIL', 'FN']]
+        ];
+    }
 }

